File: Python3/1 - 200/160 Intersection of Two Linked Lists.py
# 314ms, 99.87%
class Solution(object):
    def getIntersectionNode(self, headA, headB):
        trav_a, trav_b = headA, headB
        while trav_a is not trav_b:
            trav_a = headB if not trav_a else trav_a.next
            trav_b = headA if not trav_b else trav_b.next
        return trav_a
        

File: 1 - 200/160. Intersection of Two Linked Lists.cc
// 35ms, 66.68%
#include <bits/stdc++.h>
using namespace std;

struct ListNode {
    int val;
    ListNode *next;
    ListNode(int x) : val(x), next(NULL) {}
};

class Solution {
public:
    ListNode *getIntersectionNode(ListNode *headA, ListNode *headB) {
        ListNode *tmp1 = headA, *tmp2 = headB;
        while (tmp1 != tmp2) {
            tmp1 = tmp1 ? tmp1->next : headB;
            tmp2 = tmp2 ? tmp2->next : headA;
        }
        return tmp1;
    }
};
